BusinessLayer/InventoryCsvManager.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using Models;
+using Serilog;
+
+namespace BusinessLayer
+{
+    public class InventoryCsvManager : IInventoryManager
+    {
+        private readonly IConfigurationRoot _config;
+        private ILogger _logger;
+
+        public InventoryCsvManager(IConfigurationRoot config, ILogger logger)
+        {
+            if (!File.Exists(config["INVENTORY_PATH"])) throw new FileNotFoundException();
+
+            _config = config;
+            _logger = logger;
+        }
+
+        public Contact AddSerialNumber(Contact contact)
+        {
+            var current = GetLastSerialNumber() + 1;
+            using (var sw = new StreamWriter(_config["INVENTORY_PATH"], true))
+            {
+                var line = "\n" + current + "," + contact.Name + "," + contact.Mail + "," +
+                           DateTime.Now.ToString("dd-MM-yyyy");
+                sw.Write(line);
+            }
+
+            contact.SerialNumber = current;
+            return contact;
+        }
+
+        public long GetLastSerialNumber()
+        {
+            var reverseStream = new ReverseLineReader(_config["INVENTORY_PATH"]);
```

Comment:
new is glue, inject it

Reply:
Class ReverseLineReader is a utility class that does not implement an interface.
Do we need to create an interface for ReverseLineReader ? (This newly interface will never be implemented from another class than ReverseLineReader)